rotkehlchen/db/dbhandler.py

```diff
@@ -1747,15 +1748,31 @@ def add_manually_tracked_balances(self, data: List[ManuallyTrackedBalance]) -> N
         ) for entry in data]
         cursor = self.conn.cursor()
         try:
+            start_id = cursor.execute('SELECT MAX(id) FROM manually_tracked_balances;').fetchone()[0]  # noqa: E501
+            start_id = start_id or 0
             cursor.executemany(
                 'INSERT INTO manually_tracked_balances(asset, label, amount, location, category) '
                 'VALUES (?, ?, ?, ?, ?)', tuples,
             )
+            new_ids = [el[0] for el in cursor.execute('SELECT id FROM manually_tracked_balances WHERE id > ?;', (start_id,)).fetchall()]  # noqa: E501
         except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
             raise InputError(
                 f'One of the manually tracked balance entries already exists in the DB. {str(e)}',
             ) from e
-        insert_tag_mappings(cursor=cursor, data=data, object_reference_keys=['label'])
+
+        data_with_ids = []
```

Comment:
So I don't like this approach tbh. It seems too complicated and may fail since it depends on an auto-incrementing id which **may not be the case**. Sqlite may also reuse old delete rowids.

You can achieve the same thing in a safe way by inputting everything in a for loop at like 1753 instead of using `executemany`. Iterate through all given data, insert each one into the DB and for each cursor insertion get the `cursor.lastrowid` which should be the identifier you look for. Then edit it into the `ManuallyTrackedBalance`. Also since it's a NamedTuple I would change it to a dataclass since you need to edit it.

Reply:
Alright, changed!